tests/virt_autotest/sriov_network_card_pci_passthrough.pm

```diff
@@ -215,6 +215,13 @@ sub enable_vf {
 sub prepare_guest_for_sriov_passthrough {
     my $vm = shift;
 
+    #save original guest configuration file in case of restore in post_fail_hook()
+    my $downloaded_xml_dir = "/tmp/download_vm_xml";
+    unless (script_run("ls $downloaded_xml_dir/$vm.xml") == 0) {
+        assert_script_run "mkdir -p $downloaded_xml_dir";
+        assert_script_run "virsh dumpxml --inactive $vm > $downloaded_xml_dir/$vm.xml";
```

Comment:
Putting a newly generated guest xml to downloaded xml directory, will it make it hard to debug? Eg, when guest is restored but failed, how can people easily know whether it is restore from downloaded or the generated at the begining of test?

Reply:
if the test reuse the downloaded guest, the xml should be downloaded to the $downloaded_xml_dir, ie. since the original xml already exists, no xml will be dump here; while for the test running directly after guest installation, there is not xml, so dump it to $downloaded_xml_dir for possible restoration in post_fail_hook().